src/containers/auth/EmailPasswordSignUp.jsx

```diff
@@ -0,0 +1,141 @@
+  // Redirect to home page after it's signed up and logged in
+  if (auth.uid) {
+    return <Redirect to="/" />;
+  }
+  // Create a new account with "Email & Password" method
+  const handleSubmit = (e) => {
+    if (password === passwordRepeat) {
+      setPasswordMatch(true);
+      props.signUp({
+        firstName,
+        lastName,
+        fullName: firstName + " " + lastName,
+        email,
+        password,
+        imageURL: null,
+      });
+    } else {
+      setPasswordMatch(false);
+    }
+    e.preventDefault();
+  };
+  return (
+    <Container>
+      <h3>Creating a new account</h3>
+      <Form>
+        <Form.Row>
+          <Form.Group as={Col} controlId="fisrtName">
+            <Form.Label>First Name:</Form.Label>
+            <Form.Control
+              type="text"
+              placeholder="First name"
+              onInput={(e) => {
+                setFirstName(e.target.value);
+              }}
+            />
+          </Form.Group>
+          <Form.Group as={Col} controlId="lastName">
+            <Form.Label>Last Name:</Form.Label>
+            <Form.Control
+              type="text"
+              placeholder="Last name"
+              onInput={(e) => {
+                setLastName(e.target.value);
+              }}
+            />
+          </Form.Group>
+        </Form.Row>
+        <Form.Row>
+          <Form.Group as={Col} controlId="email">
+            <Form.Label>Email:</Form.Label>
+            <Form.Control
+              type="text"
+              placeholder="example@gmail.com"
+              onInput={(e) => {
+                setEmail(e.target.value);
+              }}
+            />
+          </Form.Group>
+          <Form.Group as={Col} controlId="password">
+            <Form.Label>Password:</Form.Label>
+            <Form.Control
+              type="password"
+              placeholder="Password"
+              onInput={(e) => {
+                setPassword(e.target.value);
+                setPasswordMatch(true);
+              }}
+            />
+          </Form.Group>
+          <Form.Group as={Col} controlId="passwordRepeat">
+            <Form.Label>Re-type password:</Form.Label>
+            <Form.Control
+              type="password"
+              placeholder="Re-type password"
+              onInput={(e) => {
+                setPasswordRepeat(e.target.value);
+                setPasswordMatch(true);
+              }}
+            />
+          </Form.Group>
+          {!passwordMatch && (
+            <p>
+              <b>Hint: </b>Those passwords don't match. Try again
+            </p>
+          )}
+        </Form.Row>
+        <Button variant="primary" type="submit" onClick={handleSubmit}>
+          Sign Up
+        </Button>
+        <Container>
+          {authError && errKey === "signup" ? (
+            <>
+              <br />
+              <p>
+                <b>{authError}</b>
+              </p>
+              <p>
+                <b>Hint: </b>
```

Comment:
usually we don't use the word hint for error messaging. perhaps google some examples of error messages